clientv3/integration/fragment_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/integration"
+	"github.com/coreos/etcd/mvcc/mvccpb"
+	"github.com/coreos/etcd/pkg/testutil"
+	"github.com/coreos/etcd/proxy/grpcproxy"
+	"github.com/coreos/etcd/proxy/grpcproxy/adapter"
+)
+
+func TestResponseWithoutFragmenting(t *testing.T) {
+	defer testutil.AfterTest(t)
+	// MaxResponseBytes will overflow to 1000 once the grpcOverheadBytes,
```

Comment:
I think another way would be to turn the (`MaxResponseBytes`)[https://github.com/coreos/etcd/blob/master/integration/cluster.go#L100] to a int instead of a uint, so that I can set its value to negative. I'm not a fan of this method, but that's the only way I can think of because adding the `grpcOverheadBytes` doesn't seem to be straightforward to (toggle)[https://github.com/mangoslicer/etcd/blob/a8c073c51e94cc4bf2dc9660f09e9f89d4d0087d/etcdserver/api/v3rpc/grpc.go#L41].